yarn-project/simulator/src/acvm/acvm.ts

```diff
@@ -27,9 +27,11 @@ type ACIRCallback = Record<
  */
 export interface ACIRExecutionResult {
   /**
-   * The partial witness of the execution.
+   * An execution result contains two witnesses.
+   * 1. The partial witness of the execution.
+   * 2. The return witness which contains the given public return values within the full witness.
    */
-  partialWitness: ACVMWitness;
+  partialAndReturnWitness: Array<ACVMWitness>;
```

Comment:
I kept the name `partialWitness` here, but I feel `solvedAndReturnWitness` would make more sense. I held off on changing it just yet, but I used "solved" rather than "partial" in acvm JS

Reply:
Could we destructure in this file the returned array? ACIR Execution Result could have partialWitness and returnWitness

Reply:
Yup can do